src/Interpreters/getOSKernelVersion.cpp

```diff
@@ -0,0 +1,31 @@
+#if defined(OS_LINUX)
+#include <Interpreters/getOSKernelVersion.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int SYSTEM_ERROR;
+}
+    
+String getOSKernelVersion()
+{
+    struct utsname os_kernel_info;
+    int buf = uname(&os_kernel_info);
+    if (buf < 0)
```

Comment:
`uname` returns error (if any), so either

```suggestion
    int err = uname(&os_kernel_info);
    if (err)
```

Or 

```suggestion
    if (uname(&os_kernel_info))
```